packages/jest-haste-map/src/HasteFS.js

```diff
@@ -42,6 +54,20 @@ export default class HasteFS {
     return this._getFileData(file) != null;
   }
 
+  follow(file: Path): Path {
+    let link;
+    for (const key of this._links.keys()) {
+      const linkRoot = path.resolve(this._rootDir, key);
+      if (file.startsWith(linkRoot + path.sep)) {
+        const cache = this._links.get(key);
+        link = cache && cache.get(fastPath.relative(linkRoot, file));
+        break;
+      }
+    }
+    if (link === undefined) return file;
+    return link[0] || (link[0] = realpath(file));
```

Comment:
The cache write here is honestly a bit hard to read. Any chance you could unroll that into slightly more verbose code that is easier to understand?